vscode-trace-extension/src/extension.ts

```diff
@@ -187,15 +189,23 @@ export function activate(context: vscode.ExtensionContext): ExternalAPI {
     return traceExtensionAPI;
 }
 
-export function deactivate(): void {
+export async function deactivate(): Promise<void> {
+    await traceServerManager.stopServer();
+    traceServerManager.dispose();
     traceLogger.disposeChannel();
     traceExtensionWebviewManager.dispose();
 }
 
-async function startTraceServerIfAvailable(): Promise<void> {
+async function startTraceServerIfAvailable(pathToTrace?: string): Promise<void> {
     const extensionId = 'vscode-trace-server';
+    if (await isUp()) {
+        return;
+    }
+    if (pathToTrace) {
```

Comment:
When using the Open Trace button in the Trace Viewer panel, the command openedTraces.openTraceFolder is invoked and this method is called without a path. The TraceServerManager will not be invoked, and the server will be started using the default tracecompass-community.vscode-trace-server command.

If the trace server manager is to be invoked in this case, the openedTraces.openTraceFolder command handler above should be changed to invoke the user dialog first, then use the chosen path as a parameter to this method.

Alternatively, the Trace Server Manager could be invoked unconditionally with the startServer() parameter changed to being optional, leading to the wildcard contributor (without validator). However I don't know if this will handle your use case because it will start the default generic trace server.

Reply:
@PatrickTasse  Thank you so much for the review!

You are correct, using "openedTraces.openTraceFolder" will lead to default server being started. I think invoking the user dialog makes more sense than the alternative. WDYT?

 Currently for us, this command is not usable since our traces are a single file trace and not a folder. We do hope to push future contributions to support opening files with that command (Out of the scope of this PR). Out of curiosity, is there any reason why it is currently designed to only open folders?

Reply:
Ah ok - Thanks for the explanation!